cpp/include/cuspatial/experimental/detail/ranges/multipoint_range.cuh

```diff
@@ -101,7 +101,8 @@ template <typename IndexType>
 CUSPATIAL_HOST_DEVICE auto multipoint_range<GeometryIterator, VecIterator>::operator[](
   IndexType idx)
 {
-  return multipoint_begin()[idx];
+  return multipoint_ref<VecIterator>{_points_begin + _geometry_begin[idx],
+                                     _points_begin + _geometry_begin[idx + 1]};
```

Comment:
I had to revert this change in #731 because I need to use this method in device code. The previous proposal requires creating a transform iterator and dereference it. Not all code path is available on device and is expensive.